src/test/java/com/alibaba/json/bvt/issue_3700/Issue3732.java

```diff
@@ -0,0 +1,67 @@
+package com.alibaba.json.bvt.issue_3700;
+
+import com.alibaba.fastjson.JSON;
+import junit.framework.TestCase;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Issue3732 extends TestCase {
+    static class Role{
+        int id;
+
+        public Role() {}
+
+        public Role(int id) {
+            this.id = id;
+        }
+
+        public void setId(int id) {this.id = id;}
+
+        public int getId() {
+            return id;
+        }
+    }
+    static class User{
+        List<Role> roleList;
+
+        public void setRoleList(List<Role> roleList) {
+            this.roleList = roleList;
+        }
+
+        public List<Role> getRoleList() {
+            return roleList;
+        }
+
+//        public void setRoleIdList(List<int> roleIdList){}
+
+        public List<Integer> getRoleIdList(){
+            if (roleList == null) return null;
+            List<Integer> roleIdList = new ArrayList<Integer>();
+            for (Role role : roleList) {
+                roleIdList.add(role.id);
+            }
+            return roleIdList;
+        }
+    }
+
+    @Test
+    public void test_for_issue(){
```

Comment:
There just have *1* test and the name of test seems meaningless.

Reply:
even after a few commits, there still only have one test method

Reply:
OK. I'll do it right now.